src/util/misc.ts

```diff
@@ -347,3 +347,9 @@ export function objToQueryString(obj: KVS<any>): string {
 export function sanitizeString(str: string) {
   return str.replace(/[&<>"'*=//]/g, ' ');
 }
+
+// We will need the encoded URL on article load so that we can set the hrefs of anchor tag correctly,
+// but we must not encode the '/' character or else relative links may fail
+export function encodeArticleIdForZimHtmlUrl(articleId: string) {
+  return encodeURIComponent(articleId).replace(/%2F/g, '/');
```

Comment:
What about:
```
return articleId && encodeURIComponent(articleId).replace(/%2F/g, '/');
```
That way we would have a better handling of the `undefined` use case. Returning `'undefined'` as string is really hiddeous.